src/models/Steps.php

```diff
@@ -198,4 +223,48 @@ private function updateBody(string $content): bool
         $req->bindParam(':item_id', $this->Entity->id, PDO::PARAM_INT);
         return $this->Db->execute($req);
     }
+
+    /*
+     * Provide SQL statement to get unfinished steps
+     *
+     * @param string $table Which table do we read from? experiments or items
+     * @param string $whereClause SQL WHERE clause
+     */
+    private function getReadAllSql(string $table, string $whereClause): string
```

Comment:
this is ugly but I know very well that sometimes it is necessary to write ugly code, and I don't have a better suggestion at the moment. Anyway, I believe in iterative code so if it works like that it can be improved later. Maybe an implementation with classes would be cleaner.